src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -274,17 +281,34 @@ public void setTargetIndex(int newTargetIdx)
                 return;
             }
 
+            RTPEncodingDesc[] sourceEncodings = sourceTrack.getRTPEncodings();
+
             int currentTL0Idx = bitstreamController.getCurrentIndex();
             if (currentTL0Idx > -1)
             {
-                currentTL0Idx = sourceTrack.getRTPEncodings()
-                    [currentTL0Idx].getBaseLayer().getIndex();
+                currentTL0Idx
+                    = sourceEncodings[currentTL0Idx].getBaseLayer().getIndex();
             }
 
             int targetTL0Idx
-                = sourceTrack.getRTPEncodings()[newTargetIdx].getBaseLayer().getIndex();
+                = sourceEncodings[newTargetIdx].getBaseLayer().getIndex();
 
-            if (currentTL0Idx != targetTL0Idx)
+            // Something lower than the current must be streaming.
+            boolean sendFIR = targetTL0Idx < currentTL0Idx;
```

Comment:
trying to make sure i'm understanding this code correctly:

1. If the target index is less than the current index, we can assume the target index stream must exist (since layers 'below' the current layer can be assumed to be sending as well
2. If the target index is greater than the current index, we need to check if target layer stream is actually sending. [if it isn't, we'll keep the target as the higher layer but fall back to the next one in other parts of the code, is that right?]

i wonder if it's just easier to do:
`sendIDR = sourceEncodings[i].isActive()` ?